tests/integration/ambient/waypoint_test.go

```diff
@@ -61,3 +67,131 @@ func TestWaypointStatus(t *testing.T) {
+				istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+					"x",
+					"waypoint",
+					"apply",
+					"--namespace",
+					nsConfig.Name(),
+					"--service-account",
+					sa,
+				})
+				retry.UntilSuccessOrFail(t, func() error {
+					fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+sa)
+					if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+						return fmt.Errorf("gateway is not ready: %v", err)
+					}
+					return nil
+				}, retry.Timeout(15*time.Second), retry.BackoffDelay(time.Millisecond*100))
+			}
+
+			output, _ := istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"list",
+				"--namespace",
+				nsConfig.Name(),
+			})
+			for _, sa := range saSet {
+				if !strings.Contains(output, sa) {
+					t.Fatalf("unexpected output: ", output)
+				}
+			}
+
+			output, _ = istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"list",
+				"-A",
+			})
+			for _, sa := range saSet {
+				if !strings.Contains(output, sa) {
+					t.Fatalf("unexpected output: ", output)
+				}
+			}
+
+			istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"-n",
+				nsConfig.Name(),
+				"delete",
+			})
+			retry.UntilSuccessOrFail(t, func() error {
+				fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"namespace")
+				// after deletion, the pod should not be found
+				if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+					if errors.Is(err, kubetest.ErrNoPodsFetched) {
+						return nil
+					}
+					return fmt.Errorf("gateway is not ready: %v", err)
+				}
+				return fmt.Errorf("gateway is not cleaned up")
+			}, retry.Timeout(15*time.Second), retry.BackoffDelay(time.Millisecond*100))
+
+			istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"-n",
+				nsConfig.Name(),
+				"delete",
+				"sa1",
+				"sa2",
+			})
+			retry.UntilSuccessOrFail(t, func() error {
+				fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"sa1")
+				// after deletion, the pod should not be found
+				if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+					if !errors.Is(err, kubetest.ErrNoPodsFetched) {
+						return fmt.Errorf("gateway is not ready: %v", err)
+					}
+				} else if err == nil {
+					return fmt.Errorf("gateway is not cleaned up")
+				}
+				fetch = kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"sa2")
+				// after deletion, the pod should not be found
+				if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+					if !errors.Is(err, kubetest.ErrNoPodsFetched) {
+						return fmt.Errorf("gateway is not ready: %v", err)
+					}
+				} else if err == nil {
+					return fmt.Errorf("gateway is not cleaned up")
```

Comment:
nit: could this error and the one above for `sa1` say something to state that it was the multi-delete that failed?  Something like `failed to remove multiple gateways at once, sa2 not cleaned up`

Reply:
This looks better, fixed.